### PR TITLE
Add network policy with user init helm chart

### PR DESCRIPTION
**Pull request by obutterbach**

### CODE REVIEW

File: charts/init-user/values.yaml [hunk 1]
```diff
@@ -4,3 +4,16 @@ Email: ""
...
+  podsMatchLabels:
+    app: nginx-ingress
+    component: controller
+
+userApps:
+  ports:
+    - protocol: TCP
+      port: 3000
+    - protocol: TCP
+      port: 8080
```
Comment (xoen) [inline charts/init-user/values.yaml]: Shouldn't we only allow traffic from the ingress controller to the auth proxy port?

Is any of the tools' auth proxy listing to port `8080`?

File: charts/init-user/templates/network-policy.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  namespace: user-{{ .Values.Username }}
+spec:
+  podSelector: {}
+  policyTypes:
+    - Ingress
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              name: {{ .Values.ingressController.namespace }}
```
[the remark this answers is not in the document]
Comment (xoen) [reply charts/init-user/templates/network-policy.yml]: Sorry that's right, it's *source* namespace which is the namespace where the ingress controller lives which I already labelled ☺️ 

Ignore me 🤦‍♂ (need a coffee 😁 )

[the remark this answers is not in the document]
Comment (obutterbach) [reply charts/init-user/templates/network-policy.yml]: aha 😄 no worries

File: charts/init-user/values.yaml (hunk 1 above)
Comment (obutterbach) [reply charts/init-user/values.yaml]: https://github.com/ministryofjustice/analytics-platform-auth-proxy/blob/master/README.md#L8
Auth-proxy listen on port `3000` and Airflow on `8080`
Do you see something else missing?

**Comment by obutterbach** [comment]
Thank you for your review 😃 
I'm planning to deploy this in `Dev` environment before we merging it here

**Comment by damacus** [comment]
@xoen whilst this is good. Unless we're going to merge and roll this out, we should delete this PR.

**Pull request closed without merging**